README.md

```diff
@@ -0,0 +1,75 @@
+# Certificate Controller
+
+## Purpose
+
+The purpose of the Certificate Controller library is to provide
+an easy way for controller authors to bootstrap webhooks while
+making it possible for users to use more customizable projects like
+[cert-manager](https://cert-manager.io/docs/) should they desire
+to do so. Its purpose is not to be a fully-featured certificate
+solution, but a stopgap that allows webhook authors to avoid
+having a hard dependency on the existence of any third-party
+certificate generation solution.
+
+This library was originally written as part of Gatekeeper to replace
+the certificate generation functionality provided by controller-runtime
+that was removed as part of a migration to Kubebuilder 2.0. It was then
+spun off as a separate repo so that other projects, such as
+[Hierarchical Namespace Controller](https://github.com/kubernetes-sigs/multi-tenancy/tree/master/incubator/hnc)
+could avoid maintaining duplicate code.
+
+## Design
+
+All behavior is governed by a `CertRotator` object that has two
+main control loops:
+
+   1. A tick-based control loop that periodically examines the certificates stored
+      in the secret and makes sure they are still valid, regenerating them if not
```

Comment:
Is it worth saying that under ordinary conditions, we expect the secrets to last for ~years?

Reply:
Maybe? I figured I'd keep things high level to give people enough context to explore the code, but not need to correct drift very often.

Reply:
sgtm